### PR TITLE
feat(brush): configure brush instance

### PR DESCRIPTION
**Pull request by cbt1**
Adds a new method to the brush instance api called `configure`. Through this method, it will be possible to configure various properties.

The first iteration will contain two properties that can control whether or not the min/max value of a range should be included when determining if a value is to be brushed.

**Example**
```js
brushInstance.configure({
   ranges: [
     { key: 'Sales' includeMax: false }, // Sales config
     { includeMax: true, includeMin: true }, // Default config
   ]
 })

brushInstance.addRange('Sales', { min: 20, max: 50 });
brushInstance.addRange('Margin', { min: 20, max: 50 });

brush.containsRangeValue('Sales', 50); // false
brush.containsRangeValue('Margin', 50); // true
```

### CODE REVIEW

File: packages/picasso.js/src/core/brush/brush.js [hunk 1]
```diff
@@ -362,6 +392,48 @@ export default function brush({
...
+    if (Array.isArray(config.ranges) && config.ranges.length) {
+      rangeConfig = {
+        sources: {},
+        default: extend({}, DEFAULT_RANGE_CONFIG)
+      };
+
+      config.ranges.forEach((cfg) => {
+        const c = {};
+        Object.keys(DEFAULT_RANGE_CONFIG)
+          .filter(key => key !== 'key')
```
Comment (miralemd) [inline packages/picasso.js/src/core/brush/brush.js]: should rename `key` to `attr` to be consistent with the next line and reduce confusion with `'key'`

**Comment by T-Wizard** [comment]
It would be nice if q-brush.js would use this to set qMinInclEq and qMaxInclEq

**Comment by cbt1** [comment]
> It would be nice if q-brush.js would use this to set qMinInclEq and qMaxInclEq

Interesting case! If it was to be automagacilly handled, then I think it would need to be tied the type of request method, which I think is out of scope for the brush.

**Comment by miralemd** [comment]
`q-brush` would only need the configuration from the `brush`, so if we could expose `brush.config` it would be fairly straight forward to implement

**Comment by cbt1** [comment]
@miralemd Still consider out of scope for this PR

**Pull request merged**